rust/agama-lib/share/profile.schema.json

```diff
@@ -251,22 +251,194 @@
-          "type": "string"
-        },
-        "lvm": {
-          "description": "Whether LVM is used.",
-          "type": "boolean"
-        },
-        "encryptionPassword": {
-          "description": "If set, the devices are encrypted using the given password.",
-          "type": "string"
+        "guided": {
+          "title": "Settings to execute a Guided Proposal",
+          "type": "object",
+          "additionalProperties": false,
+          "properties": {
+            "target": {
+              "title": "Target device",
+              "oneOf": [
+                {
+                  "type": "string",
+                  "enum": ["disk", "newLvmVg"]
+                },
+                {
+                  "type": "object",
+                  "additionalProperties": false,
+                  "properties": { "disk": { "type": "string" } },
+                  "required": ["disk"]
+                },
+                {
+                  "type": "object",
+                  "additionalProperties": false,
+                  "properties": {
+                    "newLvmVg": {
+                      "type": "array",
+                      "items": { "type": "string" }
+                    }
+                  },
+                  "required": ["newLvmVg"] 
+                }
+              ]
+            },
+            "boot": {
+              "title": "Configuration of the boot settings",
+              "type": "object",
+              "additionalProperties": false,
+              "properties": {
+                "configure": {
+                  "title": "Whether to configure partitions for booting",
+                  "type": "boolean"
+                },
+                "device": {
+                  "title": "Device to use for booting",
+                  "type": "string"
+                }
+              }
+            },
+            "encryption": {
+              "title": "Encryption settings",
+              "type": "object",
+              "additionalProperties": false,
+              "required": ["password"],
+              "properties": {
+                "password": {
+                  "title": "Passphrase to use when creating new encryption devices",
+                  "type": "string"
+                },
+                "method": {
+                  "title": "Method used to create the encryption devices",
+                  "type": "string",
+                  "enum": ["luks2", "tpm_fde"]
+                },
+                "pbkdFunction": {
+                  "title": "Password-based key derivation function to use for LUKS2",
+                  "type": "string",
+                  "enum": ["pbkdf2", "argon2i", "argon2id"]
+                }
+              }
+            },
+            "space": {
+              "title": "Policy to find space for the new partitions",
+              "type": "string",
```

Comment:
I think `type` can be omitted here.

Reply:
If it does not hurt, I think I prefer to have it explicit.

Reply:
Oh, you meant it's redundant with "enum". That's indeed true.

Reply:
Done (also in some other places).